flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/SessionReconciler.java

```diff
@@ -96,6 +96,19 @@ protected void reconcileSpecChange(
             Configuration deployConfig,
             DiffType type)
             throws Exception {
+        if (type == DiffType.SCALE) {
```

Comment:
If this logic is the same as the one in the AbstracJobReconciler we should move it instead to the shared parent class (AbstractFlinkResourceRecnciler)

Reply:
Since this case was more applicable for SessionReconciler, added the changes there. But I agree with your  point as both JobReconciler and SessionReconciler is re-using it, we can add it in the parent.